### PR TITLE
add ares to test suite (dmg only for now)

### PR DESCRIPTION
**Pull request by LukeUsher**
CGB is currently excluded as the startup_time_test phase never completes (it renders as a solid white screen)
DMG mode is working.

### CODE REVIEW

**Comment by daid** [comment]
Is CGB support planned for Ares?

File: emulators/ares.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    def getScreenshot(self):       
+        screenshot = getScreenshot(self.title_check)
+        if screenshot is None:
+            return None
+        screenshot = screenshot.reduce(3)
+        if not self.cgb:        
+            screenshot = screenshot.convert(mode="L", dither=PIL.Image.NONE)
+            screenshot = PIL.ImageOps.autocontrast(screenshot)
+            screenshot.save("tmp.png")
```
Comment (daid) [inline emulators/ares.py:46]: this will generate a lot of disk load, as getScreenshot is called a lot.

**Comment by LukeUsher** [comment]
> Is CGB support planned for Ares?

cgb is supported, and some CGB tests do pass; however the 'get-startuptime` script fails, because that particular rom hangs on a white screen in cgb mode.

File: emulators/ares.py (hunk 1 above)
Comment (LukeUsher) [reply emulators/ares.py:46]: You're right; I took this from the existing emulators/higan.py file; so didn't really think about it.

**Comment by daid** [comment]
> > Is CGB support planned for Ares?
> 
> cgb is supported, and some CGB tests do pass; however the 'get-startuptime` script fails, because that particular rom hangs on a white screen in cgb mode.

Guess you got some work then. This rom is tested on actual hardware (CGB revision E). Quite likely it's some initial value issue.

**Comment by LukeUsher** [comment]
> > > Is CGB support planned for Ares?
> > 
> > 
> > cgb is supported, and some CGB tests do pass; however the 'get-startuptime` script fails, because that particular rom hangs on a white screen in cgb mode.
> 
> Guess you got some work then. This rom is tested on actual hardware (CGB revision E). Quite likely it's some initial value issue.

I found the issue, and I do have quite some work to do: The problem is that ares does not support a CGB in DMG mode; only CGB mode.

**Comment by daid** [comment]
Oh, yes, Surprise. A DMG or a CGB in DMG mode are not the same thing. Sadly, some tests depend on CGB running in DMG mode, as some PPU timing is different.

But for passing this part, you could just fall back to DMG if the header doesn't report CGB features, it should pass that check. But it will fail a few other tests (it's what some of the lower grade emulators do)

**Pull request merged**